### PR TITLE
Add support for custom speed format.

### PR DESCRIPTION
**Pull request by wfxr**
Add support for custom speed format. The backward compatibility has been considered.

### CODE REVIEW

**Comment by ctongfei** [comment]
Thanks @wfxr ! However I would like to place new formatting options into the `ProgressBarBuilder` class instead of the main class -- can you move these as a new option there (something like `setSpeedFormat`)? I don't want to clutter the main class too much.

File: src/main/java/me/tongfei/progressbar/ProgressBarBuilder.java [hunk 1]
```diff
@@ -52,7 +53,12 @@ public ProgressBarBuilder setUnit(String unitName, long unitSize) {
     }
 
     public ProgressBarBuilder showSpeed() {
+        return showSpeed(null);
+    }
+
+    public ProgressBarBuilder showSpeed(String speedFormat) {
```
Comment (wfxr) [inline src/main/java/me/tongfei/progressbar/ProgressBarBuilder.java]: Setter for `speedFormat` has been added to `ProgressBarBuilder` already.

**Comment by wfxr** [comment]
@ctongfei Since the `ProgressThread` was created and started inner core constructor of `ProgressBar`, the `speedFormat` info must be passed into that constructor. So there is no way to add custom speed format without making modification on `ProgressBar` class.

The reason why I add a new constructor is to keep backward compatibility. Otherwise someone who use the following consturctor before will not compile anymore (Because now it requires `speedFormat`).
```java
public ProgressBar(
    String task,
    long initialMax,
    int updateIntervalMillis,
    PrintStream os,
    ProgressBarStyle style,
    String unitName,
    long unitSize,
    boolean showSpeed
)

```

File: src/main/java/me/tongfei/progressbar/ProgressBarBuilder.java (hunk 1 above)
Comment (wfxr) [reply src/main/java/me/tongfei/progressbar/ProgressBarBuilder.java]: One can pass the `speedFormat` to `showSpeed`. call `showSpeed` without argument as before is OK and the default speed format will be took.

Comment (ctongfei) [reply src/main/java/me/tongfei/progressbar/ProgressBarBuilder.java]: Got it. Thanks!

**Comment by ctongfei** [comment]
LGTM. I changed the `String` type in the builder to `DecimalFormat` (better type-safety).

**Comment by ctongfei** [comment]
Published in `0.7.3`.

**Pull request merged**